### DISCUSSION TITLE
module installation confusion

### DISCUSSION THREAD

**Discussion by morrowwm** [AllskyTeam/allsky-modules#96] [Q&A] [answered]
I'm dabbling with the module system, but am stuck getting modules installed. I've cloned the repository, and run `install.sh` to install the `allsky_dewheater` module. It shows up in `/opt/allsky/modules`.

I don't see it in the Module Manager, however. Nothing in `/var/log/allsky.log`.

I also zipped up the directory and tried using the Upload facility within the Module Manager, but got `Failed to upload the plugin. Unable to move the file`.

Version: v2023.05.01_04 on a RPi4 with 4GB RAM.

Never mind, it looks like my problems were because of some earlier efforts. I had copied a prototype module directly into `scripts/modules`, and it was generating an error in `/var/log/allskyperiodic.log`. I stopped allsky, deleted the offending module from `config/modules/postprocessing_periodic.json`, and now the Module Manager is showing the modules installed by `install.sh`. 

Leaving this here for others. I could write up some extra troubleshooting detail for the module installation step?

**Comment by Alex-developer** [comment, marked as answer]
Thanks I have logged as an issue as the module manager should handle 'broken' modules.

Thanks for the feedback

Alex

**Reply by morrowwm** [reply]
And I should use `install.sh` and expect to find my module in `/opt/allsky/modules`, correct? Not in `allsky/scripts/modules`?

**Reply by Alex-developer** [reply]
Correct. All of the modules in this repo will be installed in /opt/allsky/modules

**Reply by morrowwm** [reply]
Thank you for the confirmation, Alex.
